pkg/index/validate.go

```diff
@@ -90,8 +90,13 @@ func (p Platform) Validate() error {
 	if p.Bin == "" {
 		return errors.New("bin has to be set")
 	}
-	if len(p.Files) == 0 {
-		return errors.New("can't have a plugin without specifying file operations")
+	for _, fo := range p.Files {
```

Comment:
Let’s add a check that verifies `nil` is ok for p.Files (you should make this `Files` field a pointer like *[]FileOperaton first I think to distinguish between `nil` and empty `[]`) AND validate that empty is not allowed and should not be defaulted.